airflow/models/dag.py

```diff
@@ -928,21 +927,21 @@ def get_task_instances_before(
         corresponding to any DagRunType. It can have less if there are
         less than ``num`` scheduled DAG runs before ``base_date``.
         """
-        logical_dates: list[Any] = session.execute(
-            select(DagRun.logical_date)
+        run_after_cols: list[Any] = session.execute(
+            select(DagRun.run_after)
```

Comment:
Same, concepts like “before” and ”after” should still be defined by logical date, so we shouldn’t change this function.

Reply:
It raises the same issue. All the task instances with no logical date will not appear

Reply:
After the latest revert and testing, seems to work fine 🤔  resolve this one